makeSpectral.py

```diff
@@ -0,0 +1,259 @@
+			jucebox.append(AudioPluggin_Cpp(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "BaseAudioPlugin.h":
+			jucebox.append(AudioPluggin_H(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "BaseParameters.cpp":
+			jucebox.append(Parameters_Cpp(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "BaseParameters.h":
+			jucebox.append(Parameters_H(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "BaseProcessor.cpp":
+			jucebox.append(Processor_Cpp(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "BaseProcessor.h":
+			jucebox.append(Processor_H(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "UiContainer.cpp":
+			jucebox.append(UiContainer_Cpp(name, desc, abspath(f'{directory}/{file_path}')))
+		elif file_path == "UiContainer.h":
+			jucebox.append(UiContainer_H(name, desc, abspath(f'{directory}/{file_path}')))
+		else:
+			pass
+	return jucebox
+
+def get_args() -> ArgumentParser:
+	parser = ArgumentParser(description='Automate SpectralSuite boilerplate for new plugin.')
+	parser.add_argument('-p', '--path', type=str, nargs=1, required=True,
+               			help='The relative or absolute path to the directory containing new plugin')
+	parser.add_argument('-n', '--name', type=str, nargs=1, required=True,
+               			help='The name of the new plugin')
+	parser.add_argument('-d', '--description', type=str, nargs=1, required=False, 
+               			default=None, help='The description of the new plugin')
+	parser.add_argument('-v', '--verbose', nargs=1, required=False, default=True, 
+						type=bool, choices=[True,False])
+	return parser.parse_args()
+
+def copy_boilerplate_files_to_new_plugin(new_plugin_directory : str) -> List[str]:
+	errors = []
+	boilerplate_files = ["Base.jucer",
+					"BaseAudioPlugin.cpp",
+					"BaseAudioPlugin.h",
+					"BaseParameters.cpp",
+					"BaseParameters.h",
+					"BaseProcessor.cpp",
+					"BaseProcessor.h",
+					"UiContainer.cpp",
+					"UiContainer.h"]
+	build_directory = abspath('./SpectralSuiteBuild')
+	build_files = [f for f in listdir(build_directory)]
+	existing_files = [f for f in listdir(new_plugin_directory)]
+	for f in build_files:
+		if f in boilerplate_files  and f not in existing_files:
+			try:
+			    copy(join(build_directory, f), join(new_plugin_directory, f))
+			except SameFileError:
+			    errors.append(f"@{f} - Source and destination represents the same file.")
+			except PermissionError:
+			    errors.append(f"@{f} - Bad permission to write to the destination file.")
+			except:
+			    errors.append(f"@{f} - Error occurred while copying file.")
+	return errors
+
+def main():
+	args = get_args()
+	cmds = Cmd(args.path, args.name, args.description, args.verbose)
+
+	copy_errors = copy_boilerplate_files_to_new_plugin(cmds.path)
+	if cmds.verbose:
+		print(f"Populated {cmds.path} with boilerplate files")
+		if copy_errors:
```

Comment:
Maybe the logging on copy_errors should occurr even if cms.verbose is off? 

I think for handling the errors what you're doing at the moment is fine...just warn the user but still continue. A half created new plugin project is still valuable!

Reply:
I wondered about this too.
The errors being caught at the moment are SameFileError PermissionError.

If a SameFileError is triggered it means they *already* have the file they need in their new plugin directory.
If the user has the file but the file doesn't contain the correct text jucer.swit() will catch that and throw a ValueError.

The other use case is PermissionError which means they have the file but just have it open.
The jucer.crank() will catch that too if *and only if* that turns out to be a problem.

**However** I've also left a catch-all which I *now think is a mistake*
```python
except:
    errors.append(f"@{f} - Error occurred while copying file.")
```
If something else happens that I'm not expecting I think I'd like interpreter to throw it's hands up, halt, and tell the user.
Good spot, thanks for making me revisit this.

Reply:
Nice ok I see. Yea removing that catch-all is a good idea too. ok 👍